test/decl/func/dynamic_self.swift

```diff
@@ -24,9 +24,9 @@ enum E0 {
 class C0 {
   func f() -> Self { } // okay
 
-  func g(_ ds: Self) { } // expected-error{{covariant 'Self' can only appear as the type of a property, subscript or method result; did you mean 'C0'?}}
+  func g(_ ds: Self) { } // expected-error{{covariant 'Self' can only appear as the possibly optional type of a property, subscript or method result; did you mean 'C0'?}}
```

Comment:
@AnthonyLatsis I was wondering if this diagnostic does produces also a fix-it to rename the parameter? But by a quick look in there seems like not... Since there is a `did you mean` in there maybe having a fix-it would make sense?

Reply:
I had a good thought on this and I think the user intent is too vague to warrant a fix-it (which is supposed to actually fix the issue, rather than falling back to the closest alternative that compiles). We cannot tell whether the user really wanted to use dynamic Self or thought Self is equivalent to the enclosing type. On top of that, protocol witnesses complicate the situation because stubs want to preserve Self when in covariant position, regardless of whether the witness is supported on the adopter.

Reply:
>  We cannot tell whether the user really wanted to use dynamic Self or thought Self is equivalent to the enclosing type.

Ahh I see, that indeed makes sense. Thanks :)